pkg/tools/doc/types.go

```diff
@@ -0,0 +1,318 @@
+package doc
```

Comment:
How about moving the package to the `tools/gen` package?

Reply:
that make sense, I migrated the `generate doc` and `gen openapi` code to `tools/gen` package, and left the `command/cmd_doc` command.